services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/links/LinkMaintenanceSpec.groovy

```diff
@@ -0,0 +1,187 @@
+    int islCostWhenUnderMaintenance
+
+    def "Maintenance mode can be set/unset for a particular link"() {
+        given: "An active link"
+        def isl = topology.islsForActiveSwitches.first()
+
+        when: "Set maintenance mode for the link"
+        def response = northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, true, false))
+
+        then: "Maintenance flag for forward and reverse ISLs is really set"
+        response.each { assert it.underMaintenance }
+
+        and: "Cost for ISLs is changed respectively"
+        islUtils.getIslCost(isl) == islCostWhenUnderMaintenance + Constants.DEFAULT_COST
+        islUtils.getIslCost(islUtils.reverseIsl(isl)) == islCostWhenUnderMaintenance + Constants.DEFAULT_COST
+
+        when: "Unset maintenance mode from the link"
+        response = northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, false, false))
+
+        then: "Maintenance flag for forward and reverse ISLs is really unset"
+        response.each { assert !it.underMaintenance }
+
+        and: "Cost for ISLs is changed to the default value"
+        islUtils.getIslCost(isl) == Constants.DEFAULT_COST
+        islUtils.getIslCost(islUtils.reverseIsl(isl)) == Constants.DEFAULT_COST
+    }
+
+    def "Flows can be evacuated (rerouted) from a particular link when setting maintenance mode for it"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northbound.getAllLinks()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            allLinks.every { link ->
+                !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId)
+            } && possibleFlowPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Make the first path more preferable than others by setting corresponding costs on links"
+        possibleFlowPaths[1..-1].each { pathHelper.makePathMorePreferable(possibleFlowPaths.first(), it) }
+
+        and: "Create a couple of flows going through these switches"
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow1)
+        def flow1Path = PathHelper.convert(northbound.getFlowPath(flow1.id))
+
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow2)
+        def flow2Path = PathHelper.convert(northbound.getFlowPath(flow2.id))
+
+        assert flow1Path == possibleFlowPaths.first()
+        assert flow2Path == possibleFlowPaths.first()
+
+        and: "Reset costs back to allow rerouting flows"
+        database.resetCosts()
```

Comment:
This is a pretty flaky flow. during `pathHelper.makePathMorePreferable(possibleFlowPaths.first(), it)` you set some __link properties__ in order to change costs, but then you use __database__ to reset those costs. This leaves the actual link properties in place, but changes costs directly in db. This may lead to inconsistent behavior in the future, since your tests leave those link props.

Reply:
This also applies to cleanup steps in subsequent test